util/feeding.py

```diff
@@ -103,15 +95,12 @@ def to_sparse_tuple(sequence):
     return indices, sequence, shape
 
 
-def create_dataset(csvs, batch_size, enable_cache=False, cache_path=None, train_phase=False):
-    df = read_csvs(csvs)
-    df.sort_values(by='wav_filesize', inplace=True)
-
-    df['transcript'] = df.apply(text_to_char_array, alphabet=Config.alphabet, result_type='reduce', axis=1)
-
+def create_dataset(sources, batch_size, enable_cache=False, cache_path=None, train_phase=False, exception_box=None):
     def generate_values():
-        for _, row in df.iterrows():
-            yield row.wav_filename, to_sparse_tuple(row.transcript)
+        for sample in convert_samples(samples_from_files(sources), AUDIO_TYPE_NP):
```

Comment:
Would that source a `SDB` file and produce back `WAV` samples ?

Reply:
No - it produces `Sample` instances that carry numpy` arrays - the format the `MFCC` conversion requires.

Reply:
Ok, but those MFCC are extracted from what data, that's the part I'm lost about.

Reply:
The yield passes `sample.audio` (as  `np.float32` tensor) on to downstream code. The features will get extracted in line 84.